services/sendMail.js

```diff
@@ -1,17 +1,39 @@
 const nodemailer = require('nodemailer');
 
-const transporter = nodemailer.createTransport({
-    host: process.env.EMAIL_HOST,
-    path: process.env.EMAIL_PATH,
-    auth: {
-        user: process.env.EMAIL_ID,
-        pass: process.env.EMAIL_PASSWORD
+function createTransporter(context) {
+    let emailId, emailPassword;
+    
+    switch (context) {
+    case 'welcome':
+        emailId = process.env.WILKOMMEN_EMAIL_ID;
+        emailPassword = process.env.WILKOMMEN_EMAIL_PASSWORD;
+        console.log("its in wilkomen")
+        break;
+    case 'defect-report':
+        emailId = process.env.DEFECTREPORT_EMAIL_ID;
+        emailPassword = process.env.DEFECTREPORT_EMAIL_PASSWORD;
+        console.log("its in defect")
+        break;
+    // Add more cases
+    default:
+        throw new Error('Invalid email context provided.');
     }
-});
 
-async function sendMail(to, subject, text, html) {
+    return nodemailer.createTransport({
+        host: process.env.EMAIL_HOST, 
+        port: process.env.EMAIL_PORT ,
+        auth: {
+            user: emailId,
+            pass: emailPassword
+        }
+    });
+}
+
+async function sendMail(context = 'welcome', to, subject, text, html, attachments = []) {
```

Comment:
Parameters with default values should be at the end. Move the context to the last or second last. This way you need not even update the functions in users.js